spec/paperclip/storage/s3_spec.rb

```diff
@@ -1249,8 +1407,16 @@ class DummyCredentialProvider; end
 
     it "parses the credentials" do
       assert_equal 'pathname_bucket', @dummy.avatar.bucket_name
-      assert_equal 'pathname_key', @dummy.avatar.s3_bucket.config.access_key_id
-      assert_equal 'pathname_secret', @dummy.avatar.s3_bucket.config.secret_access_key
+
+      assert_equal 'pathname_key',
+        (defined?(::Aws) ?
+         @dummy.avatar.s3_bucket.client.config.access_key_id :
+         @dummy.avatar.s3_bucket.config.access_key_id)
+
+      assert_equal 'pathname_secret',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.